### PR TITLE
[PDI-19160]-Set variables(job entry): When "Variable substitution" is N, the value of the externally set variable is replaced.

### PR DESCRIPTION
**Pull request by krishnaalluhitachi**
Code is added to respect Variable Substitution flag so that variables declared in the Set Variables step operated as below
1. Variable will be overridden if the same variable is available in the property file and Variable Substitution flag is true
2. Variable will not be overridden if the Variable Substitution flag is false
3. Variable will be added if the variable is not available in the property file and Variable Substitution flag is true/false.

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 10m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java [hunk 1]
```diff
@@ -255,9 +255,12 @@ public Result execute( Result result, int nr ) throws KettleException {
 
       if ( variableName != null ) {
         for ( int i = 0; i < variableName.length; i++ ) {
-          variables.add( variableName[ i ] );
-          variableValues.add( variableValue[ i ] );
-          variableTypes.add( variableType[ i ] );
+          if ( ( variables.contains( variableName[ i ] ) && replaceVars ) || !variables.contains(
```
Comment (NJtwentyone) [inline engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java:258]: Is there a unit test to capture this case?

Comment (NJtwentyone) [inline engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java:258]: Is there a Testrail or Practitest to capture this case? 
Or is there been an existing Testrail or Practitest that needs to be updated?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java (hunk 1 above)
Comment (krishnaalluhitachi) [reply engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java:258]: Yes, added a new test case to cover this testJobEntrySetVariablesExecute_VARIABLE_TYPE_ROOT_JOB_VariableNotNull in JobEntrySetVariablesTest class

Comment (krishnaalluhitachi) [reply engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java:258]: Added a new test case to cover this testJobEntrySetVariablesExecute_VARIABLE_TYPE_ROOT_JOB_VariableNotNull in JobEntrySetVariablesTest class

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 25m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by krishnaalluhitachi** [comment]
Squash issue

**Pull request closed without merging**